XenonCode.hpp

```diff
@@ -471,12 +471,28 @@ const int VERSION_PATCH = 0;
 
 	// Converts a string to a double, if empty returns 0.0.
 	[[nodiscard]] inline static double ToDouble(const std::string& str) {
-		return str.empty() ? 0.0 : std::stod(str);
+		try {
+			return str.empty() ? 0.0 : std::stod(str);
+		}
+		catch (const std::invalid_argument& e) {
+			throw RuntimeError("Invalid text conversion to double");
```

Comment:
To keep it consistent with the XenonCode Runtime vocabulary, we can use Number here instead of double.